add KNET documentation and known issues (#137)
bisdn/bisdn.github.io: platform_configuration/bcm_knet_interfaces.md

---
title: Broadcom KNET network interfaces
parent: Platform Configuration

> nav_order: 3

Reviewer: ```suggestion
nav_order: 6
```

---

> Broadcom KNET network interfaces are an alternative implementation for the representation of the switch ports. They directly forward the packets in the kernel to and from the switch, reducing overhead and improving throughput and latency when communicating from the switch itself.

Reviewer: ```suggestion
Broadcom KNET network interfaces ("bcm-knet") are an alternative for the "tun" interface type representation of the switch ports. They directly forward the packets in the kernel to and from the switch, reducing overhead and improving throughput and latency when communicating from the switch itself.
```

Due to the nature of the implementation, these can only be used when running `baseboxd` on the switch itself. When running `baseboxd` off-switch, only the tap interfaces are available.

> **Note**: Broadcom KNET network interfaces is still an experimental feature, and may not always work as expected. 

Reviewer: ```suggestion
**Note**: The use of Broadcom KNET network interfaces is still an experimental feature, and may not always work as expected. 
```
